core/src/test/java/com/nextdoor/bender/ipc/IpcSenderServiceTest.java

```diff
@@ -288,8 +283,40 @@ public void testExceptionEscalation() throws TransportException, InterruptedExce
     /*
      * Expect shutdown to throw a TransportException when a child thread also throws as
      * TransportException
+     *
+     * flush() is required to be called since we added a single event only and the dummy transport
+     * buffer only empties the buffer with 5 events stored
      */
-    ipc.flush();
+    try {
+      ipc.flush();
+    } catch (TransportException e) {
+      // we expect that the factory isn't closed since the flush discovers the exception first and throws
+      verify(tfactory, never()).close();
```

Comment:
We do expect `close` to happen later right? That happens during shutdown?

Reply:
@stlava no we should not since this old unit test throws the TransportException, via the thread, when the flush() sends the remaining buffers. Once an error is found, we fail immediately instead of trying to close.